### ISSUE TITLE
Example Code

### ISSUE THREAD

**Issue by REAS** [processing/Processing-Hour-Of-Code#11]
I feel good about the programming content of the examples with the priority to keep it simple, but I think the actual visual output and the content of each program is wide open. I made two variations so far for the 3rd and 4th program. One variable focuses on moving around geometry and the second is more of a drawing program.

**Comment by shiffman** [comment]
I think the idea of a drawing program would work well in this context though I worry about the background() in setup vs. draw confusion.

Also, I understand the reason for picking ellipse() over rect() but I worry it is so limiting.  Do you think we could do both ellipse() and rect().  Can we get away with no mention of rectMode()?

**Comment by REAS** [comment]
Let's try everything you mentioned. I'll iterate the code and please do the same as you wish.

**Comment by shiffman** [comment]
Once we see the videos, I'll think we'll want to adjust these to match things I said and referred to.  This can happen later.

**Comment by scottgarner** [comment]
I'm going to assign this to Casey just to clean up the issues a bit. I like the simplicity of the current examples, but think it makes sense to add rect() if nothing else.

**Comment by scottgarner** [comment]
Current example requirements:

1) Demonstrate rect() and ellipse().
2) No alpha in colors.
3) No use of noStroke();
4) Designed for size(500,400).

**Comment by REAS** [comment]
Thanks for the short list of requirements. Will add this to my short list.

**Comment by REAS** [comment]
I'm on my third round of these, the first was simple shapes and the second was a simple robot face. I'm happiest with the third group of examples, the code is minimal and the images are partially abstract and partially representational (think Ultraman or a mask). I think each student can choose to take it one way or another. In looking at the example from other orgs using Angry Birds figures and other "robot" and "animal" themes, I started to think about the primary audience for the tutorials and the audience for Processing. I don't want our examples to be too juvenile.  Here's what I have right now:

```
// Shape

size(500, 400);

ellipse(250, 200, 300, 300);

rect(245, 40, 10, 240);

ellipse(190, 200, 70, 70);
```

```
// Color

size(500, 400);

background(10, 80, 100);

stroke(255, 255, 255);
fill(160, 220, 90);
ellipse(250, 200, 300, 300);

fill(160, 210, 230);
rect(245, 40, 10, 240);

fill(255, 255, 255);
ellipse(190, 200, 70, 70);
```

```
// Interact

void setup() {
  size(500, 400);
}

void draw() {
  background(10, 80, 100);

  stroke(255, 255, 255);
  fill(160, 220, 90);
  ellipse(mouseX, 200, 300, 300);

  fill(160, 210, 230);
  rect(245, mouseY, 10, 240);

  fill(255, 255, 255);
  ellipse(mouseX, mouseY, 70, 70);
}
```

```
// Decisions

void setup() {
  size(500, 400);
}

void draw() {
  if (mousePressed) {
    background(10, 80, 100);
  } 

  stroke(255, 255, 255);
  fill(160, 220, 90);
  ellipse(mouseX, 200, 300, 300);

  fill(160, 210, 230);
  rect(245, mouseY, 10, 240);

  fill(255, 255, 255);
  ellipse(mouseX, mouseY, 70, 70);
}
```

**Comment by REAS** [comment]
I haven't seen the final videos yet, so I don't know if Dan talked about removing background() from draw().

**Comment by scottgarner** [comment]
I like the straightforward simplicity of these and think the interact example is especially charming.

I'm pretty sure we covered background in setup, though if I remember correctly it was something we shot as an afterthought.

-S

**Comment by shiffman** [comment]
This is great.  We definitely shot video about moving background() to setup().  As with everything it was a very quick explanation.  I think it's valuable to demo that in the examples.  I would probably add an additional call to background() in setup as I believe this is how I explained it.

Anyway, I think these are awesome, and we can take another pass at them once we see the actual videos.

I've added them to the repo with 0280c925985cb416c58c1219feb3f0aa43dd9724

**Comment by shiffman** [comment]
I've also now referenced the new files in the script so they should show up in the tutorial.

5a5e6f9e229c0ba6e74edb3021571cd3411849f8

**Comment by shiffman** [comment]
The examples for shapes and color should not include size().  @scottgarner I assume this will be ok given we've solved this problem for the code that runs during the lessons themselves.

**Comment by scottgarner** [comment]
The run function should inject a size if it can't find one in the code.

**Comment by scottgarner** [comment]
I was finding the delay for loading example code to be distracting. Rather that add a loading animation or something along those lines, I've decided to just pull the PDE files directly into the page with Jekyll, which is how the hints work.

On the back end, it just means that the PDE files will be stored in _includes. I'll wait until @shiffman is done with script.js to push the changes.

**Comment by REAS** [comment]
I propose this variation to the code for 3: Interact to make it more of a painting program.

```
void setup() {
  size(500, 400);
  background(10, 80, 100);
}

void draw() {
  stroke(255, 255, 255);
  fill(160, 220, 90);
  ellipse(mouseX, 200, 300, 300);

  fill(160, 210, 230);
  rect(245, mouseY, 10, 240);

  fill(255, 255, 255);
  ellipse(mouseX, mouseY, 70, 70);
}
```

**Comment by scottgarner** [comment]
I've reworked the way the examples are loaded. Instead of loading them on the fly with Javascript, which caused a delay, they are compiled into the page source by Jekyll and are immediately available to the script.

The pde files can now be found in _includes/. @REAS, I've added the example above already.

Hopefully this isn't too confusing, but they're numbered according to the video they're associated with, so 2-5 instead of 1-4.
